### PR TITLE
Bok Boller & Chivens Spectrograph

### PR DESCRIPTION
**Pull request by mcoughlin**
I put together a branch that successfully reduces Bok Boller & Chivens Spectrograph data. I simply copied and modified one of the MMT codes, so hopefully this is consistent with expectations.

### CODE REVIEW

**Review by profxj — CHANGES_REQUESTED** [review]
This is great!

Do you have a small test suite that we can add
to our Dev Suite:  https://github.com/pypeit/PypeIt-development-suite

And a related PypeIt file.

Thanks!

**Comment by mcoughlin** [comment]
I am happy to upload files for one of the objects from a couple nights ago and pass along the pypeit file. Google drive for Macs seems to have some kind of issue, so I won't easily be able to test I guess.

**Comment by profxj** [comment]
Very good.  Please send me an email to share the Drive with you for uploading.
Thanks!

**Comment by mcoughlin** [comment]
Hopefully what I pushed is sufficient!

**Comment by profxj** [comment]
Dev suite ran successfully:

https://github.com/pypeit/PypeIt-development-suite/pull/140

Now doing full tests.

**Comment by profxj** [comment]
======================== 256 passed, 411 warnings in 1640.08s (0:27:20) ========================

**Comment by profxj** [comment]
93 passed/ 1 failed/ 0 skipped PASSED  bok_bc/600 pypeit (without masters)

The one failure is the DEIMOS one fixed by PR #1147

**Comment by profxj** [comment]
@mcoughlin -- Can you add the bok to a few of the docs:

`install.rst`

and the `README.md`

All set otherwise.  Thanks!

**Comment by profxj** [comment]
And, sorry, I see one more issue:

Please Edit the PR (click the button at the top)
to redirect it to `develop` instead of `release`.

Thanks!

**Comment by mcoughlin** [comment]
@profxj I think I added it to the expected lines. Lemme know.

**Comment by mcoughlin** [comment]
@profxj Can you help me understand what to do about that CHANGES.rst issue?

**Comment by profxj** [comment]
fixed (I think)

**Review by profxj — APPROVED** [review]
all good now!

File: pypeit/core/fitting.py
Comment (jhennawi) [inline pypeit/core/fitting.py]: Why was this change made? I am not sure I agree with it. There is no fit if maskwork has no True elements, and so a crash with this error message is probably desired.

File: pypeit/spectrographs/bok_bc.py
Comment (jhennawi) [inline pypeit/spectrographs/bok_bc.py]: Please change the docs to indicate this is Bok and not WHT/ISIS.

**Review by jhennawi — APPROVED** [review]

**Review by jhennawi — CHANGES_REQUESTED** [review]
Minor but important comment to biplane. Not sure if that change was made in this PR, but this is the first I'm seeing it.

File: pypeit/core/fitting.py
Comment (mcoughlin) [reply pypeit/core/fitting.py]: I had an object for which this line was crashing the pipeline, and I saw other failure modes also simply returned and then went with a different estimation technique. It seemed to allow things to pass OK. Obviously it is easy enough to revert, I had forgotten about it.

File: pypeit/spectrographs/bok_bc.py
Comment (mcoughlin) [reply pypeit/spectrographs/bok_bc.py]: This I have fixed. Just let me know what to do with the previous one.

File: pypeit/core/fitting.py
Comment (jhennawi) [reply pypeit/core/fitting.py]: Sorry for the delay here. Please revert this for now. Regarding the crash, it would be great if you could post an issue with where you think there is a bug with a link to the minimal pypeit file required to reproduce it, and the data.

**Comment by jhennawi** [comment]
Ready to approve after: 

1) B-spline change is reverted
2) Updated develop needs to be pulled in (@profxj or @kbwestfall maybe you can help with that in case there are the usual conflicts to resolve). 

I think we are good with the tests, so after that I'll approve.

File: pypeit/core/fitting.py
Comment (mcoughlin) [reply pypeit/core/fitting.py]: This is reverted now.

**Comment by mcoughlin** [comment]
@jhennawi I have reverted the B-spline change. If @profxj wouldn't mind doing the final rebase, I would appreciate it. It looks like we are now a bit behind the develop branch.

**Comment by profxj** [comment]
did my part. Back to you @jhennawi

**Review by jhennawi — APPROVED** [review]

**Comment by jhennawi** [comment]
Thanks for this great PR!

**Pull request merged**
